vint.go

```diff
@@ -60,3 +60,46 @@ func mustReadByte(r io.Reader) (byte, error) {
 
 	return b, err
 }
+
+func WriteVInt(w io.Writer, n int64) (error) {
+	if n >= -112 && n <= 127 {
+		_, err := w.Write([]byte{byte(int8(n))})
+		return err
+	}
+
+	len := -112
+	if n < 0 {
+		n = n ^ -1
+		len = -120
+	}
+
+	tmp := n
+	for tmp != 0 {
+		tmp = tmp >> 8
+		len--
+	}
+
+	_, err := w.Write([]byte{byte(int8(len))})
```

Comment:
We should be able to do this in a single write. We know the number of bytes here, I think, so we can allocate a slice and then index into it.